analysis/generate_report.py

```diff
@@ -143,6 +144,40 @@ def get_experiment_data(experiment_names, main_experiment_name,
     return experiment_df, description
 
 
+def filter_experiments(all_experiment_names):
+    """Helper function that reads data from disk or from the database. Returns
+    a list of old experiments of the same type as the main experiment."""
+    main_experiment_name, prev_experiment_names = all_experiment_names[
+        0], all_experiment_names[1:]
+    logger.debug('Verifying benchmark type of the main experiment...')
+    experiment_benchmark_df = queries.get_experiment_benchmarks(
+        [main_experiment_name])
+    main_experiment_type = experiment_utils.get_experiment_type(
+        experiment_benchmark_df['benchmark'].unique().tolist())
+    logger.info('Main experiment benchmark type is %s.', main_experiment_type)
+
+    same_type_experiments = [main_experiment_name]
+    for experiment_name in prev_experiment_names:
```

Comment:
This does a lot of unnecessary SQL queries. Can we rethink this so that we get the data for all experiments at once?
In SQL the query looks something like `SELECT DISTINCT benchmark, experiment FROM trial WHERE ...` just convert this to the right SQL alchemy instead of doing a new query per trial.

Reply:
Yep, that is a good point. I worried about the same issue when writing the `for` loop, but I did not think of a better solution at that time.
How about this:
I guess we only care about the benchmarks used in the `main_experiment`, so instead of filtering experiment names here, I will add another filter to only include those benchmarks when [querying the experiment data](https://github.com/google/fuzzbench/blob/cb748014dc3f1da4b4ac01c4c435b09ff6858ea5/analysis/generate_report.py#L140).

Reply:
Here is [the latest sample](https://storage.googleapis.com/www.fuzzbench.com/reports/2022-11-30-16-33-15m-ldg/index.html), which includes old results of the same benchmarks as in the current experiments only.
I reckon this is also more reasonable, because if we did not run the new fuzzer on a benchmark, then we cannot compare the new fuzzer with old results on that benchmark.

Reply:
@jonathanmetzman : A friendly ping on checking the new logic above to filter benchmarks. 
Here is the implementation: [Only merge with the same benchmark programs](https://github.com/google/fuzzbench/pull/1562/commits/73478a9c4d099b56aae58a2cf314b190278f8f6a).
The other comments are addressed in [Minor fixes on log and code style](https://github.com/google/fuzzbench/pull/1562/commits/4226bed505775be1b027b740ba3a742c3c36030b).
Thanks!